cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -204,8 +224,8 @@ func defaultPromptValue(locationParam azure.ArmTemplateParameterDefinition) *str
 	return nil
 }
 
-func (p *BicepProvider) LastDeployment(ctx context.Context) (*azapi.ResourceDeployment, error) {
-	modulePath := p.modulePath()
+func (p *BicepProvider) LastRootDeployment(ctx context.Context) (*azapi.ResourceDeployment, error) {
```

Comment:
I feel like this bicep provider implementation has been getting unwieldy for a while.  So many different cases and edge cases happening in here now and might be time for a general refactor.

Some idea:
- Have a 1:1 between calls to the bicep provider and bicep deployment
- Perhaps composability is a new infra provider that then consumes the bicep provider after the bicep is generated.

Reply:
Agreed that  a refactor for bicepModule is long overdue. I plan to tackle this after landing the changes, since it will help inform the shape of the contracts between bicepProvider and bicepModule.

From the changes that are being applied here, there isn't behavior specific to composability other than the new `appModules()` method. Composability plays within the rules of Bicep provider, and not the other way around.

Reply:
I suspect that for this particular change of `LastDeployment` -> `LastRootDeployment`,  we expect the "DeploymentService" (which also includes VS) to be passing a modulePath as the parameter -- which is an API breaking change, but it's kept as `LastRootDeployment` to maintain backwards compatibility and limit scope. I can write a follow-up issue to delegate that responsibility downwards.